server/cluster/diagnosis.go

```diff
@@ -0,0 +1,85 @@
+package cluster
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/tikv/pd/pkg/cache"
+	"github.com/tikv/pd/server/schedule/checker"
+	"github.com/tikv/pd/server/schedule/operator"
+	"github.com/tikv/pd/server/schedule/plan"
+	"github.com/tikv/pd/server/statistics"
+)
+
+const maxDiagnosisResultNum = 6
+
+// diagnosisManager is used to manage diagnose mechanism which shares the actual scheduler with coordinator
+type diagnosisManager struct {
+	cluster      *RaftCluster
+	schedulers   map[string]*scheduleController
+	checkers     *checker.Controller
+	dryRunResult map[string]*cache.FIFO
+}
+
+func newDiagnosisManager(cluster *RaftCluster, schedulerControllers map[string]*scheduleController, checkers *checker.Controller) *diagnosisManager {
+	return &diagnosisManager{
+		cluster:      cluster,
+		schedulers:   schedulerControllers,
+		checkers:     checkers,
+		dryRunResult: make(map[string]*cache.FIFO),
+	}
+}
+
+func (d *diagnosisManager) diagnosisDryRun(name string) error {
+	var result *diagnosisResult
+	// if it's a scheduler.
+	_, ok := d.schedulers[name]
+	if ok {
+		result = newDiagnosisResult(d.schedulers[name].DiagnoseDryRun())
+	}
+	// if it's a checker.
+	switch name {
+	case "merge-checker":
+		regions := d.cluster.GetRegionStatsByType(statistics.UndersizedRegion, 100)
```

Comment:
If it do not have a `region` info, check all regions with `UndersizedRegion` should be ok, otherwise we should consider how to show the rest regions result==

Reply:
Or, maybe we could provide an API to get `GetRegionStatsByType`, and the region id was needed for dry-run for checker.